app/test_main.py

```diff
@@ -1 +1,73 @@
-# write your code here
+import pytest
+from unittest import mock
+
+
+import app.main
+
+
+@pytest.mark.parametrize(
+    "valid_google_url_value,has_internet_connection_value,expectation",
+    [
+        pytest.param(
+            False,
+            True,
+            "Not accessible",
+            id="Should test if the url is valid"
+        ),
+        pytest.param(
+            True,
+            False,
+            "Not accessible",
+            id="Should test if the we have internet connection"
+        ),
+        pytest.param(
+            True,
+            True,
+            "Accessible",
+            id="Should test if the url is valid"
+        ),
+        pytest.param(
+            False,
+            False,
+            "Not accessible",
+            id="Should test both two functions"
+        )
+    ]
+)
+@mock.patch("app.main.valid_google_url")
+@mock.patch("app.main.has_internet_connection")
+def test_access_google_page(
+    mock_has_internet_connection: mock.MagicMock,
+    mock_valid_google_url: mock.MagicMock,
+    valid_google_url_value: bool,
+    has_internet_connection_value: bool,
+    expectation: str
+) -> None:
+    mock_valid_google_url.return_value = valid_google_url_value
+    mock_has_internet_connection.return_value = has_internet_connection_value
+    assert (
+        app.main.can_access_google_page("https://www.google.com")
+        == expectation
+    )
+
+
+@mock.patch("app.main.valid_google_url")
```

Comment:
would make sense to test these functions with different param combinations - you will get different results if `has_internet_connection` function returns True or False

Reply:
This solution is not optimal - you repeat the same test 4 times, instead of making one test with 4 different sets of params - same as `test_access_google_page`